### ISSUE TITLE
package: userpath

### ISSUE THREAD

**Issue by InLaw** [cript0nauta/pynixify#36]
pyspark installation

```
dask
pyct
rasterio
numba
arcgis
datashape
ipython 
jupyterlab
geopandas
pipx
pip
networkx
pyspark

```

after 
nix-shell shell.nix 

```
Requirement already satisfied: click in /nix/store/02mkfzlja90abihab4fxg2kwz9z8aly7-python3.7-click-7.0/lib/python3.7/site-packages (from userpath==1.4.1) (7.0)
ERROR: Could not find a version that satisfies the requirement distro; platform_system == "Linux" (from userpath==1.4.1) (from versions: none)
ERROR: Could not find a version that satisfies the requirement distro; platform_system == "Linux" (from userpath==1.4.1) (from versions: none)
ERROR: No matching distribution found for distro; platform_system == "Linux" (from userpath==1.4.1)
ERROR: No matching distribution found for distro; platform_system == "Linux" (from userpath==1.4.1)
Processing ./py4j-0.10.7-py2.py3-none-any.whl
Processing ./py4j-0.10.7-py2.py3-none-any.whl
builder for '/nix/store/b69f6lxiln7451cbys16f52ymmzq5y7r-python3.7-userpath-1.4.1.drv' failed with exit code 1
cannot build derivation '/nix/store/qz5vijmxgx04wia2ajbsvsr6wda944ly-python3-3.7.6-env.drv': 1 dependencies couldn't be built
error: --- Error --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix-shell
build of '/nix/store/qz5vijmxgx04wia2ajbsvsr6wda944ly-python3-3.7.6-env.drv' failed


```

without pyspark:
nix-shell shell.nix 
```
Processing ./userpath-1.4.1-py3-none-any.whl
Requirement already satisfied: click in /nix/store/02mkfzlja90abihab4fxg2kwz9z8aly7-python3.7-click-7.0/lib/python3.7/site-packages (from userpath==1.4.1) (7.0)
Requirement already satisfied: click in /nix/store/02mkfzlja90abihab4fxg2kwz9z8aly7-python3.7-click-7.0/lib/python3.7/site-packages (from userpath==1.4.1) (7.0)
ERROR: Could not find a version that satisfies the requirement distro; platform_system == "Linux" (from userpath==1.4.1) (from versions: none)
ERROR: Could not find a version that satisfies the requirement distro; platform_system == "Linux" (from userpath==1.4.1) (from versions: none)
ERROR: No matching distribution found for distro; platform_system == "Linux" (from userpath==1.4.1)
ERROR: No matching distribution found for distro; platform_system == "Linux" (from userpath==1.4.1)
builder for '/nix/store/b69f6lxiln7451cbys16f52ymmzq5y7r-python3.7-userpath-1.4.1.drv' failed with exit code 1
cannot build derivation '/nix/store/lkl94wm0zk9vlnpa9sy5p868fhkzd9qc-python3-3.7.6-env.drv': 1 dependencies couldn't be built
error: --- Error --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix-shell
build of '/nix/store/lkl94wm0zk9vlnpa9sy5p868fhkzd9qc-python3-3.7.6-env.drv' failed

```

**Comment by cript0nauta** [comment]
The cause of the problem is the `userpath` package, which is a dependency of pipx. I'll investigate the issue. In the meantime, you can remove pipx from your requirements.

**Comment by cript0nauta** [comment]
userpath also uses `extras_require`: https://github.com/ofek/userpath/blob/master/setup.py#L54. This is related to #44.

**Comment by cript0nauta** [comment]
Related to this:

- https://github.com/pypa/pip/pull/2153
- https://stackoverflow.com/questions/16055403/setuptools-platform-specific-dependencies/16055603#16055603

**Comment by cript0nauta** [comment]
Fixed in 4f0c35e453324d23ee70357baea052538f0f872c. Thanks for noting this!
